merkle/hashers/tree_hasher.go

```diff
@@ -42,3 +48,47 @@ type MapHasher interface {
 	// TODO(gbelvin): Replace Size() with BitLength().
 	Size() int
 }
+
+var (
+	maxHashers = trillian.HashStrategy(len(trillian.HashStrategy_value))
+	logHashers = make([]LogHasher, maxHashers)
+	mapHashers = make([]MapHasher, maxHashers)
+)
+
+// RegisterLogHasher registers a hasher for use.
+func RegisterLogHasher(h trillian.HashStrategy, f LogHasher) {
+	if h < 0 || h >= maxHashers {
+		panic(fmt.Sprintf("RegisterLogHasher(%v) of unknown hasher", h))
+	}
+	logHashers[h] = f
+}
+
+// RegisterMapHasher registers a hasher for use.
+func RegisterMapHasher(h trillian.HashStrategy, f MapHasher) {
+	if h < 0 || h >= maxHashers {
+		panic(fmt.Sprintf("RegisterMapHasher(%v) of unknown hasher", h))
+	}
+	mapHashers[h] = f
+}
+
+// NewLogHasher returns a LogHasher
+func NewLogHasher(h trillian.HashStrategy) (LogHasher, error) {
```

Comment:
Turns out `LogHasher` conflicts with the interface name. 
Even though this isn't technically create new objects (because there's no need to), it seems like it fits in the New* category because it's like a factory method.  What do you think?